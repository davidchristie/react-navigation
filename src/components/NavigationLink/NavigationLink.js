import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { NavLink } from 'react-router-dom'

export default class NavigationLink extends Component {
  static propTypes = {
    to: PropTypes.string.isRequired
  }

  render () {
    return (
      <li className='NavigationLink'>
        <NavLink exact to={this.props.to}>
          {this.props.children}
        </NavLink>
      </li>
    )
  }
}
