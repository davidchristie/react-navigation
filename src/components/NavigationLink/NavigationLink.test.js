import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'

import NavigationLink from './NavigationLink'

it('renders correctly', () => {
  const props = {
    to: 'link'
  }
  expect(toJson(shallow(
    <NavigationLink {...props}>
      Content
    </NavigationLink>
  ))).toMatchSnapshot()
})
