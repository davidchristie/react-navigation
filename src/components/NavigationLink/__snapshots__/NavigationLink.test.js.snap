// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="NavigationLink"
>
  <NavLink
    activeClassName="active"
    exact={true}
    to="link"
  >
    Content
  </NavLink>
</li>
`;
